Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -0,0 +1,140 @@
+        /// characters or less.  It must only contain letters (a-z, A-Z), numbers (0-9), or underscores (_).
+        /// </para>
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = "builder")]
+        public string DatasetId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The "friendly name" field for the dataset.  Used for a descriptive name for the dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = "builder")]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The "description" field for the dataset.  Used for a user-friendly description of the dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = "builder")]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The default lifetime for tables in the dataset to exist in ms.  Minimum is 3600000 (1hr).  
+        /// When the expiration time is reached, the table will be automatically deleted.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = "builder")]
+        public int Timeout { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            Dataset newData;
+            // Check whether they are passing in a completed Dataset object or are building one with params.
+            if (Dataset != null)
+            {
+                newData = Dataset;
+            }
+            // Builder mode selected.
+            else
+            {
+                // Check for required fields.
+                if (DatasetId == null)
+                {
+                    WriteError(new ErrorRecord(new Exception("DatasetId missing."), "Insert failed.",
+                        ErrorCategory.InvalidArgument, DatasetId));
+                    return;
+                }
+                // Everything present.  Create a new Dataset object.
+                newData = new Dataset();
+                newData.FriendlyName = (Name != null) ? Name : "New Dataset";
+                newData.Description = (Description != null) ? Description : "New Dataset";
+                newData.DefaultTableExpirationMs = Timeout;
+                newData.DatasetReference = new DatasetReference();
+                newData.DatasetReference.DatasetId = DatasetId;
+                newData.DatasetReference.ProjectId = Project;
+            }
+
+            // Add the new dataset to the project supplied.
+            DatasetsResource.InsertRequest request = Service.Datasets.Insert(newData, Project);
+            Dataset response = request.Execute();
+            if (response != null)
+            {
+                WriteObject(response, true);
```

Comment:
We should just use WriteObject(response). WriteObject(response, true) is if the response is an Enumerable.